front/components/ConnectorPermissionsModal.tsx

```diff
@@ -629,6 +631,24 @@ export function ConnectorPermissionsModal({
   }
 
   async function save() {
+    // confirmation in case there are private nodes
+    const privateNodes = Object.values(selectedNodes).filter(
+      (node) => node.node.providerVisibility === "private"
+    );
+
+    if (privateNodes.length > 0) {
+      const confirmed = await confirm({
+        title: "Sensitive data synchronization",
+        message: `You are synchronizing data from ${privateNodes
```

Comment:
Should we say something like "You are synchronizing data from those private channels:"?  I think it may be confusing for people if they pick a mix of private an non-private and only see confirmation for some channels.

Also, what if there are 30 private channels selected ? This modal would not really work right ?